src/zope/session/session.py

```diff
@@ -33,48 +35,24 @@
         IClientIdManager, IClientId, ISession, ISessionDataContainer, \
         ISessionPkgData, ISessionData
 
-try:
-    from cStringIO import StringIO
-except ImportError:
-    # Py3: Now StringIO location.
-    from io import StringIO
 
 try:
     from UserDict import IterableUserDict as UserDict
 except ImportError:
-    # Py3: Now StringIO location.
+    # Py3: New UserDict location.
     from collections import UserDict
 
-try:
-    from base64 import encodebytes
-except ImportError:
-    # Py3: Python 2 only has encodestring
-    from base64 import encodestring as encodebytes
-
-try:
-    from threading import get_ident
-except ImportError:
-    # Py3: Python 2 has a different location.
-    try:
-        from thread import get_ident
-    except ImportError:
-        # Python 3.2 has yet a different location
-        from threading import current_thread
-        def get_ident():
-            return current_thread().ident
-
-try:
-    unicode
-except NameError:
-    # Py3: Define unicode
-    unicode = str
+_PY3 = bytes is not str
+encodebytes = base64.encodebytes if _PY3 else base64.encodestring
+get_ident = __import__('threading' if _PY3 else 'thread').get_ident
```

Comment:
I'm not in love with this trick.

Reply:
I won't claim it's *good* code, but it does ensure coverage parity across platforms. In limited amounts, in projects that don't need a `_compat.py` (like this one now!) I don't mind it---I prefer it over '#pragma: no cover". But if there are strong objections or a better way, I'm happy to change it!

Reply:
An alternative would be `from six.moves._thread import get_ident`. That function exists natively on 2.7 and 3.x. However, that has two problems, one minor, one not:

- This project doesn't list a dependency on `six` (it may already be getting one transitively). That's the minor one.
- That would bypass any monkey patching that projects like gevent and eventlet do, which only touch the python module AFAICS. That seems major.

Reply:
We use the

```
try:
    # Python 3
    from X import foo
except ImportError:
    # Python 2
    from Y import foo
```
pattern elsewhere, and it seems strange to switch to `__import__(...)` for just one import.  That's all I meant; the rest looks fine.

Reply:
That pattern either leads to incomplete coverage on one platform, or requires a "#pragma: no cover" for the alternate branch. Right now, this project has *zero* "#pragma: no cover" directives (except for the oblig in the package `__init__.py`, which may not be necessary anymore, actually). That's why I did it. 

I can understand that it might be bothersome. If you'd rather see a "#pragma", I will happily switch it. I have to make some changes to this branch anyway.

Reply:
The only reason it gets away with zero pragmas is that you're measuring coverage on Python 3, and all the 'except ImportError' have the Python 2 branch in the try: part and the Python 3 branch in the except: part.  :-)

All I want is consistency with Python-version-conditional imports used in the same file (for, e.g., `UserDict`), and no unnecessary `__import__()` statements.

Reply:
You're absolutely right. I didn't see that block above, and with it there, in retrospect I don't know why I did this one this way. Will change.

Reply:
Thanks for your patience.